url-shortener/src/main/java/com/bobocode/shortener/service/DefaultURLShortenService.java

```diff
@@ -0,0 +1,45 @@
+package com.bobocode.shortener.service;
+
+import com.bobocode.shortener.dto.URLPayload;
+import com.bobocode.shortener.entity.ShortenedUrl;
+import com.bobocode.shortener.repository.ShortenedUrlRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Service
+@RequiredArgsConstructor
+public class DefaultURLShortenService implements URLShortenService {
+
+    private static final String URL_TEMPLATE = "boboshort.";
+    private final ShortenedUrlRepository shortenedUrlRepository;
+
+    @Transactional
+    @Override
+    public String shortenUrl(final URLPayload payload) {
+        Objects.requireNonNull(payload);
+        final int randomValue = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+        var url = URL_TEMPLATE + randomValue;
+        var shortenedUrl = ShortenedUrl.builder()
+                .shortUrl(url)
+                .originalUrl(payload.getUrl())
+                .title(payload.getTitle())
+                .createdAt(LocalDateTime.now())
+                .build();
+        this.shortenedUrlRepository.save(shortenedUrl);
+        return url;
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public String findOriginalURLByShortUrl(final String shortUrl) {
```

Comment:
@vromantsev caching is crucial for URL shorteners, so it makes sense to enable caching for this method.

Reply:
added caching